### PR TITLE
Add responsible-disclosure.md

### LINKED ISSUE #1489: Review our security policy
We should think about how we want to deal with security disclosures for zebra and to document this procedure.

Right now we just point people to security@zfnd.org.

### PR DESCRIPTION
**Pull request by dconnolly**
## Motivation

Publicly commit to our security disclosure policy.

## Solution

Add a responsible-disclosure.md that lists our bilateral disclosure agreement with the ECC and soon our public PGP key for reporting security issues to us.

## Review

Not super urgent; @teor2345 has been part of this discussion.

## Related Issues

#1489

## Follow-on work

https://github.com/ZcashFoundation/zebra/issues/1638

### CODE REVIEW

File: responsible_disclosure.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+Zcash is a technology that provides strong privacy. Notes are encrypted to their destination, and then the monetary base is kept via zero-knowledge proofs intended to only be creatable by the real holder of Zcash. If this fails, and a counterfeiting bug results, that counterfeiting bug might be exploited without any way for blockchain analyzers to identify the perpetrator or which data in the blockchain has been used to exploit the bug. Rollbacks before that point, such as have been executed in some other projects in such cases, are therefore impossible.
+
+The standard describes reporters of vulnerabilities including full details of an issue, in order to reproduce it. This is necessary for instance in the case of an external researcher both demonstrating and proving that there really is a security issue, and that security issue really has the impact that they say it has - allowing the development team to accurately prioritize and resolve the issue.
+
+In the case of a counterfeiting bug, however, just like in CVE-2019-7167, we might decide not to include those details with our reports to partners ahead of coordinated release, so long as we are sure that they are vulnerable.
+
+
```
Comment (teor2345) [inline responsible_disclosure.md]: Nit: Extra newline
```suggestion
```

File: responsible_disclosure.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+Zcash is a technology that provides strong privacy. Notes are encrypted to their destination, and then the monetary base is kept via zero-knowledge proofs intended to only be creatable by the real holder of Zcash. If this fails, and a counterfeiting bug results, that counterfeiting bug might be exploited without any way for blockchain analyzers to identify the perpetrator or which data in the blockchain has been used to exploit the bug. Rollbacks before that point, such as have been executed in some other projects in such cases, are therefore impossible.
+
+The standard describes reporters of vulnerabilities including full details of an issue, in order to reproduce it. This is necessary for instance in the case of an external researcher both demonstrating and proving that there really is a security issue, and that security issue really has the impact that they say it has - allowing the development team to accurately prioritize and resolve the issue.
+
+In the case of a counterfeiting bug, however, just like in CVE-2019-7167, we might decide not to include those details with our reports to partners ahead of coordinated release, so long as we are sure that they are vulnerable.
```
Comment (teor2345) [inline responsible_disclosure.md]: Do we need to change this text because we didn't make the disclosure decision for CVE-2019-7167 ?

Comment (dconnolly) [reply responsible_disclosure.md]: Correct yes

File: responsible_disclosure.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
```
Comment (dconnolly) [inline responsible_disclosure.md]:
```suggestion
In the case of a counterfeiting bug, we might decide not to include those details with our reports to partners ahead of coordinated release, so long as we are sure that they are vulnerable.
```

**Comment by dconnolly** [comment]
Fixed some wording from @teor2345 's review, merging but open to more comments/discussion.

**Pull request merged**